src/algorithms/calorimetry/CalorimeterHitReco.cc

```diff
@@ -234,7 +234,7 @@ void CalorimeterHitReco::process(
         std::vector<double> cdim;
         // get segmentation dimensions
         auto segmentation_type = m_converter->findReadout(local).segmentation().type();
-        if (segmentation_type == "CartesianGridXY") {
+        if (segmentation_type == "CartesianGridXY" || segmentation_type == "HexGridXY") {
```

Comment:
Can we use typeid for this instead of hit-by-hit string comparisons that differ only in the final two letters for some cases? E.g. without the `HexGridXY` yet this looks like: 
```diff
diff --git a/src/algorithms/calorimetry/CalorimeterHitReco.cc b/src/algorithms/calorimetry/CalorimeterHitReco.cc
index 9214b319..7c313c18 100644
--- a/src/algorithms/calorimetry/CalorimeterHitReco.cc
+++ b/src/algorithms/calorimetry/CalorimeterHitReco.cc
@@ -17,6 +17,9 @@
 #include <DD4hep/Volumes.h>
 #include <DD4hep/config.h>
 #include <DDSegmentation/BitFieldCoder.h>
+#include <DDSegmentation/CartesianGridXY.h>
+#include <DDSegmentation/NoSegmentation.h>
+#include <Evaluator/DD4hepUnits.h>
 #include <Evaluator/DD4hepUnits.h>
 #include <Math/GenVector/Cartesian3D.h>
 #include <Math/GenVector/DisplacementVector3D.h>
@@ -233,16 +236,16 @@ void CalorimeterHitReco::process(
         const auto pos = local.nominal().worldToLocal(gpos);
         std::vector<double> cdim;
         // get segmentation dimensions
-        auto segmentation_type = m_converter->findReadout(local).segmentation().type();
-        if (segmentation_type == "CartesianGridXY") {
+        auto& segmentation_type = typeid(m_converter->findReadout(local).segmentation());
+        if (segmentation_type == typeid(dd4hep::DDSegmentation::CartesianGridXY)) {
             auto cell_dim = m_converter->cellDimensions(cellID);
             cdim.resize(3);
             cdim[0] = cell_dim[0];
             cdim[1] = cell_dim[1];
             m_log->debug("Using segmentation for cell dimensions: {}", fmt::join(cdim, ", "));
         } else {
-            if ((segmentation_type != "NoSegmentation") && (!warned_unsupported_segmentation)) {
-                m_log->warn("Unsupported segmentation type \"{}\"", segmentation_type);
+            if ((segmentation_type != typeid(dd4hep::DDSegmentation::NoSegmentation)) && (!warned_unsupported_segmentation)) {
+                m_log->warn("Unsupported segmentation type \"{}\"", segmentation_type.name());
                 warned_unsupported_segmentation = true;
             }
 
```

Reply:
implemented your proposed changes. thanks.